posthog/hogql_queries/web_analytics/web_overview_pre_aggregated.py

```diff
@@ -0,0 +1,142 @@
+from typing import TYPE_CHECKING
+from datetime import datetime, UTC
+
+from posthog.hogql import ast
+from posthog.hogql.parser import parse_select
+
+if TYPE_CHECKING:
+    from posthog.hogql_queries.web_analytics.web_overview import WebOverviewQueryRunner
+
+
+class WebOverviewPreAggregatedQueryBuilder:
+    # Supported property keys and their mapping to pre-aggregated table fields
+    SUPPORTED_PROPERTIES = {
+        "$host": "host",
+        "$device_type": "device_type",
+    }
+
+    def __init__(self, runner: "WebOverviewQueryRunner") -> None:
+        self.runner = runner
+
+    def can_use_preaggregated_tables(self) -> bool:
```

Comment:
Do we preaggregate by `host X device_type` together? As in, cartesian product? Or just individual filters? Because if we arent doing cartesian, then we need to check this here

Reply:
So, the way it works on those tables is that it yields the same result as the group by results from the query, which is nice. However, we have to know which pre-aggregated table to choose for each query, as there won't be a one-size-fits-all solution.

For our case here, we're ok. We can combine the pre-aggregated rows by host/device type, so the results are good as long as we use any dimension that exists in the tables.